tests/PySys/log_request_end_to_end/log_request_error_messages/run.py

```diff
@@ -43,20 +43,22 @@ def validate(self):
 
     @retry(Exception, tries=20, delay=1)
     def wait_until_logs_retrieved(self):
-       
+
         log_file = self.cumulocity.retrieve_log_file(self.operation_id)
         if len(log_file) != 0:
             return self.download_file_and_verify_error_messages(log_file)
         else:
             raise Exception("retry")
-      
+
     def create_logs_for_test(self):
-        log = self.startProcess(
-            command=self.sudo,
-            arguments=[
-                "python3", f"{os.getcwd()}/log_request_end_to_end/create_test_logs.py"],
-            stdouterr="log_failed",
-        )
+        # remove if there are any old files
+        os.system("sudo rm -rf /tmp/sw_logs")
+        
+        #create logs
+        os.system(f"{os.getcwd()}/log_request_end_to_end/create_test_logs.py")
```

Comment:
I'd have exposed the log generation logic as a function which could be directly invoked from here so that it doesn't need to be executed as a script here. That function could take all the inputs that it needs like the seed value, line count etc and return the temp file path of the generated file.

That will eliminate the possibility of the log generation script executing with a different python than the one that we're running these tests with.

Reply:
Yes, done.